src/System.IO.Ports/src/System/IO/Ports/SerialStream.cs

```diff
@@ -957,7 +954,12 @@ public unsafe override int EndRead(IAsyncResult asyncResult)
             // Free memory, GC handles.
             NativeOverlapped* overlappedPtr = afsar._overlapped;
             if (overlappedPtr != null)
-                Overlapped.Free(overlappedPtr);
+            {
+                // Legacy behavior as indicated by tests (e.g.: System.IO.Ports.Tests.SerialStream_EndRead.EndReadAfterClose)
+                // expects to be able to call EndRead after Close/Dispose - even if disposed _threadPoolBinding can free the
+                // native overlapped.
+                _threadPoolBinding.FreeNativeOverlapped(overlappedPtr);
```

Comment:
The code below the comment throws if EndWrite or EndBegin was called more then once so the code never reaches the point of trying to release it twice.